workspaces/mcp-chat-client/plugins/mcp-chat-client-backend/src/services/MCPClientServiceImpl.ts

```diff
@@ -0,0 +1,349 @@
+            ];
+          } catch (error) {
+            throw new Error(
+              `Failed to find npx for server '${serverConfig.name}': ${
+                error instanceof Error ? error.message : error
+              }. Please ensure Node.js is properly installed with npx available.`,
+            );
+          }
+        } else if (serverConfig.scriptPath) {
+          // Use script path
+          const isPythonScript = serverConfig.scriptPath.endsWith('.py');
+          const isWindows = process.platform === 'win32';
+
+          if (isPythonScript) {
+            command = isWindows ? 'python' : 'python3';
+          } else {
+            command = process.execPath;
+          }
+          args = [serverConfig.scriptPath, ...(serverConfig.args || [])];
+        } else {
+          throw new Error(
+            `Server config for '${serverConfig.name}' must have either scriptPath, npxCommand, or url`,
+          );
+        }
+
+        transport = new StdioClientTransport({
+          command,
+          args,
+          env: {
+            ...process.env, // Inherit current environment
+            ...serverConfig.env, // Add config-specific env vars
+            // Ensure node is in PATH when using npx
+            ...(serverConfig.npxCommand && {
+              PATH: `${path.dirname(process.execPath)}:${
+                process.env.PATH || ''
+              }`,
+            }),
+          },
+        });
+
+        console.log(
+          `Connecting to MCP server '${serverConfig.name}' via STDIO`,
+        );
+      }
+
+      // Connect the client with the appropriate transport
+      await client.connect(transport);
+
+      const toolsResult = await client.listTools();
+
+      const serverTools = toolsResult.tools.map(tool => ({
+        type: 'function',
+        function: {
+          name: tool.name,
+          description: tool.description,
+          parameters: tool.inputSchema,
+        },
+        serverId: serverConfig.name, // Track which server this tool belongs to
+      }));
+
+      allTools.push(...serverTools);
+      this.mcpClients.set(serverConfig.name, client);
+
+      console.log(
+        `MCP Server '${serverConfig.name}' connected with tools:`,
+        serverTools.map(t => t.function.name),
+      );
+    }
+
+    this.tools = allTools;
+    this.connected = true;
+    console.log(
+      'All MCP servers connected. Total tools:',
+      this.tools.map(t => t.function.name),
+    );
+  }
+
+  async processQuery(
+    messagesInput: any[],
+    enabledTools: string[] = [],
+  ): Promise<{
+    reply: string;
+    toolCalls: ToolCall[];
+    toolResponses: any[];
+  }> {
+    const messages: ChatMessage[] = [
+      {
+        role: 'system',
+        content:
+          "You are a helpful assistant. When using tools, provide a clear, readable summary of the results rather than showing raw data. Focus on answering the user's question with the information gathered.",
```

Comment:
is this a golden prompt that makes everything work or should be customizable by the adopters?

Reply:
Yes, this is a golden prompt that works across all use cases.
Based on all my tests so far, it has consistently delivered results. In fact, it worked so well that I had completely forgotten about it until now!